app/code/core/Mage/CatalogRule/Model/Action/Index/Refresh.php

```diff
@@ -116,9 +116,7 @@ public function execute()
     {
         $this->_app->dispatchEvent('catalogrule_before_apply', array('resource' => $this->_resource));
 
-        /** @var Mage_Core_Model_Date $coreDate */
```

Comment:
My bad, it's incorrect to use `storeTimeStamp()`, `$timestamp` must be GMT (same timezone as in the DB)  and not based on local timezone. So it should be:

```suggestion
        /** @var Mage_Core_Model_Date $coreDate */
        $coreDate  = $this->_factory->getModel('core/date');
        $timestamp = $coreDate->gmtTimestamp();
```

The above `$timestamp` has the same value as the original PR `Mage::app()->getLocale()->date(null, null, null, true)->get(Zend_Date::TIMESTAMP)`.

The original `$timestamp = $coreDate->gmtTimestamp('Today');` returns today's date at 0 hours and not the current time. See https://github.com/OpenMage/magento-lts/blob/cc3eff3ebec46efbcab7ae4c26dd8a754476da18/app/code/core/Mage/Core/Model/Date.php#L142-L150

Reply:
I love how Magento has 99 ways to get a timestamp.. Isn't the default timezone already UTC? If so is that not the same as `time()`?

If it just uses the current timestamp is the result of the index refresh dependent on the time of day that it is run? Sorry, I haven't dug into this issue deeply enough to actually understand what the bug is and what the fix should be..

Reply:
Related to this commit I discovered issues with Magento installations, but I did not check in OM yet. Even I configured correctly the time zones in PHP and Magento, always the time is not correct in frontend, logs. For example, the special price period should end at 00:00 but it ends at 2 am. In logs when there are errors/warnings the time is not correct. There is a 2 - 3 hours difference related to normal time. When I check webserver logs I need to make a conversion in my mind.